File: Aviasales/framework/browser.js
require('chromedriver');
require('geckodriver');
const {Builder} = require('selenium-webdriver');
const config = require('../config.json');
const logger = require('./utils/log.util');

class Browser {

    constructor() {
        this.driver;
    }

    async setUpDriver() {
        logger.info('Browser is started');
        try {
            switch (config.browser) {
                case 'chrome':
                    this.driver = new Builder().forBrowser('chrome').build();
                    break;
                case 'firefox':
                    this.driver = new Builder().forBrowser('firefox').build();
                    break;
                default:
                    logger.info("Select browser chrome/firefox");
                    break;
            }
            await this.driver.get(config.startURL);
            await this.driver.manage().window().maximize();
            await this.driver.manage().setTimeouts(config.timeouts);
        } catch (error) {
            logger.error(`Cannot start browser: ${error}`);
        }
    }

    async quit() {
        logger.info("Quit from driver");
        await this.driver.quit().catch((error) => {
            logger.warning(`Error during closing the browser: ${error}`);
        });
    }

    async refresh() {
        logger.info("Refresh page");
        await this.driver.navigate().refresh().catch((error) => {
            logger.warning(`Error during closing the browser: ${error}`);
        });
    }

    async getScreenshot() {
        return await this.driver.takeScreenshot().catch((error) => {
            logger.warning(`${error}`);
        });
    }

    async navigate(url) {
        logger.info(`Navigate to ${url}`);
        await this.driver.get(url).catch((error) => {
            logger.warning(`Error during closing the browser: ${error}`);
        });
    }
}

module.exports = Browser;

File: ButtleShip/src/main/kotlin/framework/browser/Browser.kt
package framework.browser

import framework.utils.Logger
import framework.utils.getParameter
import io.github.bonigarcia.wdm.WebDriverManager
import org.openqa.selenium.WebDriver
import org.openqa.selenium.chrome.ChromeDriver
import org.openqa.selenium.firefox.FirefoxDriver
import java.util.concurrent.TimeUnit

object Browser {

    private var driver: WebDriver? = null
    private var logger: Logger = Logger()

    fun getInstance() {
        val browser: String? = getParameter("browser")
        logger.info("Driver $browser init")
        driver = when (browser) {
            "chrome" -> {
                WebDriverManager.chromedriver().setup()
                ChromeDriver()
            }
            "firefox" -> {
                WebDriverManager.firefoxdriver().setup()
                FirefoxDriver()
            }
            else -> throw IllegalBrowserNameExceptions()
        }
    }

    fun getDriver(): WebDriver? {
        return driver
    }

    fun closeBrowser() {
        logger.info("Close driver")
        driver?.quit()
        driver = null
    }

    fun setImplicitlyWait(timeout: String?) {
        logger.info("Set implicitly wait $timeout seconds")
        driver?.manage()?.timeouts()?.implicitlyWait(timeout!!.toLong(), TimeUnit.SECONDS)
    }

    fun setMaxSizeWindow() {
        logger.info("Set max size window")
        driver?.manage()?.window()?.maximize()
    }

    fun setUrl(url: String?) {
        logger.info("Set URL $url")
        driver?.get(url)
    }
}